test/e2e/framework/framework.go

```diff
@@ -93,6 +93,11 @@ type TestDataSummary interface {
 type FrameworkOptions struct {
 	ClientQPS   float32
 	ClientBurst int
+
+	// Tie the framework to a specific node (e.g. for node e2e tests).
+	// NOTE: This must be a pointer because the node-name flag has not been parsed yet when the
+	// framework options are set.
```

Comment:
To reviewers:
I'm not satisfied with this solution, but don't have a better idea that doesn't require a significant refactoring. If you have any ideas, I'm all ears 👂👂!